extensions/amp-story-audio-sticker/0.1/amp-story-audio-sticker.js

```diff
@@ -24,26 +100,120 @@ export class AmpStoryAudioSticker extends AMP.BaseElement {
+        StateProperty.CURRENT_PAGE_ID,
+        (currentPageId) =>
+          this.toggleStickerState_('on-active-page', pageId === currentPageId)
+      );
+    });
+  }
+
+  /** @private */
+  maybeInitializeDefaultSticker_() {
+    this.defaultSticker_ = this.getDefaultSticker_();
+    if (!this.defaultSticker_) {
+      return;
+    }
+
+    // Remove all existing child elements and add only pretap and posttap elements.
+    this.element.replaceChildren();
+    this.element.appendChild(
+      <amp-story-audio-sticker-pretap>
+        <amp-img
+          width={DEFAULT_STICKERS[this.defaultSticker_].width}
+          height={DEFAULT_STICKERS[this.defaultSticker_].height}
+          layout="responsive"
+        ></amp-img>
+      </amp-story-audio-sticker-pretap>
+    );
+    this.element.appendChild(
+      <amp-story-audio-sticker-posttap>
+        <amp-img
+          width={DEFAULT_STICKERS[this.defaultSticker_].width}
+          height={DEFAULT_STICKERS[this.defaultSticker_].height}
+          layout="responsive"
+        ></amp-img>
+      </amp-story-audio-sticker-posttap>
+    );
+  }
+
+  /**
+   * Return the specified default sticker or null if there's custom sticker provided.
+   * Return fallback sticker is both default and custom sticker config are not specified at all.
+   * @private
+   * @return {string}
+   */
+  getDefaultSticker_() {
+    if (
+      scopedQuerySelector(this.element, STICKER_PRETAP_TAG) &&
+      scopedQuerySelector(this.element, STICKER_POSTTAP_TAG)
+    ) {
+      return null;
+    }
+
+    const stickerAttr = this.element.getAttribute('sticker');
+    return stickerAttr || FALLBACK_DEFAULT_STICKER;
+  }
+
+  /**
+   * Toggle sticker states.
+   * @param {string} stateName
+   * @param {boolean} force
+   * @private
+   */
+  toggleStickerState_(stateName, force) {
+    this.mutateElement(() => this.element.classList.toggle(stateName, force));
+  }
+
+  /** @private */
+  initializeListeners_() {
+    this.element.addEventListener(
+      'click',
+      () => this.storeService_.dispatch(Action.TOGGLE_MUTED, false),
+      true
+    );
+    // TODO: add listeners for click animations.
   }
 
   /** @override */
   layoutCallback() {
-    return Services.storyStoreServiceForOrNull(this.win).then(
-      (storeService) => {
-        this.element.addEventListener('click', () =>
-          storeService.dispatch(Action.TOGGLE_MUTED, false)
-        );
-      }
-    );
+    if (this.defaultSticker_) {
+      const pretapImage = scopedQuerySelector(
+        this.element,
+        'amp-story-audio-sticker-pretap amp-img'
+      );
+      pretapImage.setAttribute(
+        'src',
+        DEFAULT_STICKERS[this.defaultSticker_].pretapUrl
+      );
+      const posttapImage = scopedQuerySelector(
+        this.element,
+        'amp-story-audio-sticker-posttap amp-img'
+      );
+      posttapImage.setAttribute(
+        'src',
+        DEFAULT_STICKERS[this.defaultSticker_].posttapUrl
+      );
```

Comment:
Can you do this in buildCallback as well? You should be able to set them in the same place as you create them.

Reply:
I can but I intentionally move them to here. Shouldn't the layoutCallback be the place you put the real resource loading logic according to here: https://github.com/ampproject/amphtml/blob/main/docs/building-an-amp-extension.md#layoutcallback?

Reply:
`layoutCallback` is the place where you would call `resource.load()` but you don't need to do that with images because they load immediately.

The problem with the current approach is that you might change the attribute after `amp-img` read it, so the URL never gets applied to the `<img>` inside `amp-img`. If you create the element at the same time that you apply the `src` value, it's more guaranteed to work.

Reply:
Also, if you move this to the creation, you'd be able to remove the variable `this.defaultSticker` since it would only be needed locally.